### ISSUE TITLE
Exporting Revit walls to IfcCoveringType

### ISSUE THREAD

**Issue by AndersAlverfeldt** [Autodesk/revit-ifc#155]
Is there any way to export Revit walls to IfcCoveringType using the IFCExportType and IFCExportAs parameters? No matter what parameter values I set, the walls export to IfcWallStandardCase. This is working as expected for, for example, Revit ceilings and floors but not for walls or curtain walls.
![IfcCoveringType](https://user-images.githubusercontent.com/59955162/72509131-33bbca80-3847-11ea-802f-445810626811.png)

**Comment by tamder** [comment]
Any updates on this? It is an issue for at least 5 years, and it prevents us from providing appropriate IFCs, and makes us appologize to the customers every time.

**Comment by AngelVelezSosa** [comment]
We are looking at similar issues at the moment, so can try to get this in also.

**Comment by tamder** [comment]
I've found that most of the system families have this non-documented hardcoded behaviour, which makes it very difficult to export proper IFC-s, since in most projects we cannot stick to the OOTB Revit Categories (so to speak miss-use them) because their graphical and other particular behaviour.
At the and of the day the most convenient solution would be to get rid of the IfcExportAs limitations completely and let the user to export a Revit Railing as and IfcCoveringType.SKIRTINGBOARD if thats the correct class for that element.
Honestly, I think adusting, or fine-tuning these System Family related hardcoded mappings is a dead-end.

**Comment by AngelVelezSosa** [comment]
Agreed, and that is the approach we are taking at least for walls and railings as a start.  There is sadly no such thing as a System Family (a system family is basically something that doesn't have a real family element), but we plan to allow exporting a wall as anything.
... although a recent bug report shows that "anything" has its issues.  Really, it will be a child of IfcProduct that's allowed.

**Comment by tamder** [comment]
By "System Families" I mean anything that is not an instance of FamilyInstance API class. System Families have their own API classes, that can cause the confusion, but I think you are aware of that. Anyway, I'm glad to hear that you are taking that approach. Children of IfcProduct is quite enough.

**Comment by thomvill** [comment]
Any update on this? As far as I can tell, there is still no functionality for exporting ceiling grid lines to IFC, so curtain wall are often used as substitute.

**Comment by AngelVelezSosa** [comment]
You should be able to export a wall as covering types - are you finding this is not the case?

**Comment by thomvill** [comment]
I'm finding this is not the case-- the curtain wall (drawn as Revit Roof), comes out as IfcRoof and IfcCurtainWallType in Solibri.

I'm also having trouble modifying the IFC Export AS parameter in Revit, after updating to the latest IFC Exporter today (v23.3.0.0). That is likely an issue on my end, however.

![image](https://github.com/Autodesk/revit-ifc/assets/115538666/93f08b84-aeb1-410c-ab67-46a3e8f49b11)

**Comment by AngelVelezSosa** [comment]
"Export to IFC as" is the new parameter to use.  And OK, the issue is that a _roof_ can't be exported as a covering type?  That would be a different (if related) issue.

**Comment by ArtemHnatko** [comment]
Issue is Not Reproduced in Revit 2023-2025: Walls correctly export as IfcCoveringType when the IFCExportAs and IFCExportType parameters are set. The expected classification is applied in the IFC output, indicating that the issue does not occur in this version.

**Testing environment:**

- Autodesk Revit 2023 (v.23.1.60.0)
- Autodesk Revit 2024
- Autodesk Revit 2025

![Image](https://github.com/user-attachments/assets/02ff0f6a-d8cd-4df9-8229-d00c055c4c67)
